src/spatio_temporal_voxel_grid.cpp

```diff
@@ -200,10 +204,12 @@ void SpatioTemporalVoxelGrid::TemporalClearAndGenerateCostmap(                \
         if (time_until_decay < 0.)
         {
           // expired by acceleration
+          cleared_point = true;
           if(!this->ClearGridPoint(pt_index))
           {
             std::cout << "Failed to clear point." << std::endl;
           }
+          break;
```

Comment:
remove this break, I believe I was wrong here. What should be happening is that for each clearing sensor it should be able to accelerate the points to handle multiple-overlapping frustums to allow more significant clearing. 

This is a really subtle function that I haven't been developing on for a few years now, apologies for some of the flip-flopping.

Reply:
That `break` only happens when the point is cleared from the grid due to its decay time being less than 0 (it's inside the `if (time_until_decay < 0.)` statement). In this case, there is no need to check other frustums -- even if another overlapping frustum would further accelerate the decay, the point was already cleared from the grid. I think your initial reasoning for adding this `break` statement is correct.

Reply:
:facepalm: clearly I haven't looked at this in depth recently.